apps/site/assets/ts/schedule/components/schedule-finder/UpcomingDepartures.tsx

```diff
@@ -34,91 +142,216 @@ export const RoutePillSmall = ({
+            <CrTableRow
+              scheduledStops={liveTripData.by_trip[tripId].schedules}
+            />
+          )}
+        />
+      ))
+    );
+  }
+  return null;
+};
+
+export const UpcomingBusDepartures = ({
+  tripData,
+  routeId,
+  directionId,
+  stopId
+}: UpcomingBusProps): ReactElement<HTMLElement> | null => {
+  const initialState: PredictionState = {
+    data: null,
+    initial: null,
+    isLoading: true,
+    error: false
+  };
+  const [predictionState, predictionDispatch] = useReducer(
+    reducer,
+    initialState
+  );
+
+  useEffect(
+    () => {
+      fetchPredictionData(routeId, stopId, directionId, predictionDispatch);
+    },
+    [routeId, directionId, stopId]
+  );
+
+  const {
+    data: predictions,
+    error: predictionError,
+    isLoading: arePredictionsLoading
+  }: PredictionState = predictionState;
+
+  if (!predictions || predictionError || arePredictionsLoading) {
     return null;
   }
-  if (predictions !== null && hasPredictions(predictions)) {
-    return (
-      <>
-        <h3>Upcoming Departures</h3>
-        <table className="schedule-table">
-          <thead className="schedule-table__header">
-            <tr className="schedule-table__row-header">
-              <th>Destinations</th>
-            </tr>
-          </thead>
-          <tbody>
-            {predictions.map((prediction: StopPrediction, idx: number) => (
-              <TableRow
-                prediction={prediction}
-                // eslint-disable-next-line react/no-array-index-key
-                key={idx}
-              />
-            ))}
-          </tbody>
-        </table>
-      </>
+
+  return (
+    <UpcomingBusDepartureRows tripData={tripData} predictions={predictions} />
+  );
+};
+
+export const UpcomingBusDepartureRows = ({
+  tripData,
+  predictions
+}: {
+  tripData: ServiceScheduleInfo;
+  predictions: StopPrediction[];
+}): ReactElement<HTMLElement> | null => {
+  if (predictions && hasBusPredictions(predictions)) {
+    const liveTripData = TripDataForPredictions(tripData, predictions);
+    return wrapDepartures(
+      predictions.map((prediction: StopPrediction) => (
+        <Accordion
+          key={prediction.trip_id}
+          trip={liveTripData.by_trip[prediction.trip_id]}
+          contentCallback={() => <BusTableRow prediction={prediction} />}
+        />
+      ))
     );
   }
   return null;
 };
 
+export const UpcomingDepartures = ({
+  tripData,
+  routeId,
+  directionId,
+  stopId
+}: Props): ReactElement<HTMLElement> | null => {
+  if (!tripData) return null;
+
+  const allTripNames = tripData.trip_order;
+  const allTrips = tripData.by_trip;
+  const firstTrip = allTripNames[0];
+  const firstStopSchedule = allTrips[firstTrip];
+  const firstScheduledStop = firstStopSchedule.schedules[0];
+  const mode = firstScheduledStop.route.type;
+
+  return mode === 2 ? (
+    <UpcomingCrDepartures tripData={tripData} />
+  ) : (
+    <UpcomingBusDepartures
```

Comment:
We seem to be assuming that any mode which isn't commuter rail should be treated as buses. If this component is only intended to be used with those two modes, could we have a fallback that errors out or fails gracefully if the mode is something unexpected?